### PR TITLE
Doc: Add tests for the datagen webhook js docs widget

### PR DESCRIPTION
**Pull request by bobbyiliev**
### Motivation

Adding some basic tests for the webhooks datagen docs JS widget.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Review by nrainer-materialize — APPROVED** [review]

File: ci/test/docs-widgets/webhooks-datagen.test.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  });
+
+  it("should initialize the DOM elements correctly", async () => {
+    const webhookURLInput = window.document.getElementById("webhookURL");
+    const authPasswordInput = window.document.getElementById("authPassword");
+    const jsonSchemaTextarea = window.document.getElementById("jsonSchema");
+    const startButton = window.document.getElementById("startButton");
+    const stopButton = window.document.getElementById("stopButton");
+
+    await new Promise((r) => setTimeout(r, 650));
```
Comment (RobinClowers) [inline ci/test/docs-widgets/webhooks-datagen.test.js:38]: https://vitest.dev/api/vi.html#vi-waitfor should make this test more reliable and probably faster in many cases.

File: ci/test/docs-widgets/webhooks-datagen.test.js [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    const stopButton = window.document.getElementById("stopButton");
+    const webhookURLInput = window.document.getElementById("webhookURL");
+    const authPasswordInput = window.document.getElementById("authPassword");
+
+    webhookURLInput.value = "http://localhost";
+    authPasswordInput.value = "password";
+
+    startButton.click();
+
+    await new Promise((r) => setTimeout(r, 500));
```
Comment (RobinClowers) [inline ci/test/docs-widgets/webhooks-datagen.test.js:72]: Probably all of these timeouts could be `waitFor` instead.

**Review by RobinClowers — APPROVED** [review]
Looks good, just one suggestion!

File: ci/test/docs-widgets/webhooks-datagen.test.js (hunk 1 above)
Comment (bobbyiliev) [reply ci/test/docs-widgets/webhooks-datagen.test.js]: Oh nice! Thanks for that suggestion Robin!

**Pull request merged**